packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -0,0 +1,164 @@
+	 * ```
+	 */
+	public setTemplate(template: MessageEmbedOptions | MessageEmbed) {
+		this.embedTemplate = template instanceof MessageEmbed ? template : new MessageEmbed(template);
+		return this;
+	}
+
+	/**
+	 * Sets a format callback that will be mapped to each embed field in the array of items when the embed is paginated. This should convert each item to a format that is either text itself or can be serialized as text.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginatedFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginatedFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 * @param value The formatter callback to be applied to each embed item
+	 */
+	public formatItem(item: (item: T, index: number, array: T[]) => any) {
+		this.items = this.items.map(item);
+		return this;
+	}
+
+	/**
+	 * Build the pages of the given array.
+	 *
+	 * You must call the [[PaginatedFieldMessageEmbed.make]] and [[PaginatedFieldMessageEmbed.run]] methods last, in that order, for the pagination to work.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginatedFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginatedFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 */
+	public make() {
+		if (!this.fieldTitle.length) throw new Error('The title of the field to format must have a value.');
+		if (!this.items.length) throw new Error('The items array is empty.');
+		if (this.items.some((x) => !x)) throw new Error('The format of the array items is incorrect.');
+
+		this.totalPages = Math.ceil(this.items.length / this.itemsPerPage);
+		this.generatePages();
+		return this;
+	}
+
+	private generatePages() {
+		const template = this.embedTemplate instanceof MessageEmbed ? (this.embedTemplate.toJSON() as MessageEmbedOptions) : this.embedTemplate;
+		for (let i = 0; i < this.totalPages; i++) {
+			const clonedTemplate = new MessageEmbed(template);
+			const fieldsClone = this.embedTemplate.fields;
+			clonedTemplate.fields = [];
+
+			if (!clonedTemplate.color) clonedTemplate.setColor('RANDOM');
```

Comment:
hmm..what if color is 0x000000 :megu:

Reply:
Well, that color is set for all embedded pages, it doesn't depend on my code, it depends on discord.js. :bruh:

Reply:
@vladfrangu RANDOM is a valid ColorResolvable